src/displayBackend/drm/DrmDeviceDetector.cpp

```diff
@@ -0,0 +1,162 @@
+/*
+ *  DRM Detector helper module
+ *
+ *   This program is free software; you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program; if not, write to the Free Software
+ *   Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307 USA
+ *
+ * Copyright (C) 2020 EPAM Systems Inc.
+ *
+ */
+#include "DrmDeviceDetector.hpp"
+
+#include <memory>
+#include <string>
+#include <stdexcept>
+
+#include <libudev.h>
+#include <xf86drm.h>
+#include <xf86drmMode.h>
+#include <fcntl.h>
+
+#include <xen/be/Log.hpp>
+
+namespace Drm {
+
+struct drmFdGuard {
```

Comment:
Just to do not break project code style: could we have drmFdGuard upper case?

Reply:
please specify the name, I am not quite sure I get who this should look like all in upper case ;)